elastic/beats
# [Heartbeat] Retry on no previous state

## Proposed commit message



Add "no previous state found" condition to state transition to trigger retries. With this change, hearbeat will attempt a retry (if configured to do so) not only when the previous state is "up" but also when it wasn't possible to find a previous state.

## Checklist




## How to test this PR locally



1. Build heartbeat locally.
2. Configure a monitor with a non-previously existing ID and `maxAttempts > 1`,point it somewhere to get a "down" result.
3. Run HB and check a retry is triggered on the first down result.
